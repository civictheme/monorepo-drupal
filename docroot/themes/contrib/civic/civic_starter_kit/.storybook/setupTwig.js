const twigDrupal = require('twig-drupal-filters');

/**
 * Configures and extends a standard twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
