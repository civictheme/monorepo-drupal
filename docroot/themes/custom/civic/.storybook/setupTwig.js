const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');

module.exports.namespaces = {
  atoms: resolve(__dirname, '../', 'components/01-atoms'),
  molecules: resolve(__dirname, '../', 'components/02-molecules'),
  organisms: resolve(__dirname, '../', 'components/03-organisms'),
  templates: resolve(__dirname, '../', 'components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
