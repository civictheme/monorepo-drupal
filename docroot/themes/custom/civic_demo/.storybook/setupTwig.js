const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');


module.exports.namespaces = {
  atoms: resolve(__dirname, '../', '.storybook-components/01-atoms'),
  molecules: resolve(__dirname, '../', '.storybook-components/02-molecules'),
  organisms: resolve(__dirname, '../', '.storybook-components/03-organisms'),
  templates: resolve(__dirname, '../', '.storybook-components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
