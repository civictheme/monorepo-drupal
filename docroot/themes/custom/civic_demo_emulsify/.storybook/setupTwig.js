const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');
const twigBEM = require('bem-twig-extension');
const twigAddAttributes = require('add-attributes-twig-extension');

module.exports.namespaces = {
  atoms: [
    resolve(__dirname, '../../civic_emulsify', 'components/01-atoms'),
    resolve(__dirname, '../', 'components/01-atoms'),
  ],
  molecules: [
    resolve(__dirname, '../../civic_emulsify', 'components/02-molecules'),
    resolve(__dirname, '../', 'components/02-molecules'),
  ],
  organisms: [
    resolve(__dirname, '../../civic_emulsify', 'components/03-organisms'),
    resolve(__dirname, '../', 'components/03-organisms'),
  ],
  templates: [
    resolve(__dirname, '../../civic_emulsify', 'components/04-templates'),
    resolve(__dirname, '../', 'components/04-templates'),
  ]
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigBEM(twig);
  twigDrupal(twig);
  twigAddAttributes(twig);
  return twig;
};
