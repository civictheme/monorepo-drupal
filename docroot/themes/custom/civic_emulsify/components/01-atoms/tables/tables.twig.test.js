import { join } from 'path';
import { render, Twig } from 'twig-testing-library';

import { setupTwig } from '../../../.storybook/setupTwig';

setupTwig(Twig);

describe('tables', () => {
  it('can render a table', async () => {
    const { container } = await render(join(__dirname, 'tables.twig'));

    expect(container).toMatchSnapshot();
  });
});
