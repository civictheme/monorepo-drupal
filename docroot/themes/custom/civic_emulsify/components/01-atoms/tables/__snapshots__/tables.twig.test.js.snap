// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tables can render a table 1`] = `
<div>
  

  <table
    class="table"
  >
    
	
    <thead>
      
		
      <tr>
        
			
        <td
          class="table__heading-cell"
          colspan="4"
        >
          This is a heading
        </td>
        
		
      </tr>
      
	
    </thead>
    
	
    <tbody>
      
  	
      <tr
        class="table__row"
      >
        
  		
        <td
          class="table__cell"
        >
          Row 1
        </td>
        
  		
        <td
          class="table__cell"
          colspan="2"
        >
          Information about something
        </td>
        
  		
        <td
          class="table__cell"
        >
          And this is important
        </td>
        
  	
      </tr>
      
  	
      <tr
        class="table__row"
      >
        
      
        <td
          class="table__cell"
        >
          Row 2
        </td>
        
  		
        <td
          class="table__cell"
          colspan="2"
        >
          Information about something
        </td>
        
  		
        <td
          class="table__cell"
        >
          And this is important
        </td>
        
  	
      </tr>
      
  	
      <tr
        class="table__row"
      >
        
      
        <td
          class="table__cell"
        >
          Row 3
        </td>
        
  		
        <td
          class="table__cell"
          colspan="2"
        >
          Information about something
        </td>
        
  		
        <td
          class="table__cell"
        >
          And this is important
        </td>
        
  	
      </tr>
      
  	
      <tr
        class="table__row"
      >
        
      
        <td
          class="table__cell"
        >
          Row 4
        </td>
        
  		
        <td
          class="table__cell"
          colspan="2"
        >
          Information about something
        </td>
        
  		
        <td
          class="table__cell"
        >
          And this is important
        </td>
        
  	
      </tr>
      
	
    </tbody>
    
	
    <tfoot>
      
		
      <tr>
        
			
        <td
          class="table__heading-cell table__footer-cell"
          colspan="4"
        >
          This is a table footer
        </td>
        
		
      </tr>
      
	
    </tfoot>
    

  </table>
  

</div>
`;
